packages/loopring_v3/contracts/operator/Operator.sol

```diff
@@ -0,0 +1,94 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity 0.5.7;
+
+import "../lib/Ownable.sol";
+
+import "../iface/IExchange.sol";
+
+/// @title Operator
+/// @author Daniel Wang  - <daniel@loopring.org>
+contract Operator is Ownable
+{
+    address  exchange;
+
+    function commitBlock(
+        uint8 blockType,
+        uint16 blockSize,
+        bytes calldata data
+        )
+        onlyOwner
+        external
+        returns (bytes32 merkleRootAfter)
+    {
+       return IExchange(exchange).commitBlock(blockType, blockSize, data);
+    }
+
+    function verifyBlock(
+        uint blockIdx,
+        uint256[8] calldata proof
+        )
+        external
+        onlyOwner
+    {
+        IExchange(exchange).verifyBlock(blockIdx, proof);
+    }
+
+    function revertBlock(
+        uint blockIdx
+        )
+        external
+        onlyOwner
+    {
+        IExchange(exchange).revertBlock(blockIdx);
+    }
+
+    function withdrawBlockFee(
+        uint blockIdx,
+        address payable feeRecipient
+        )
+        external
+        onlyOwner
+        returns (uint feeAmount)
+    {
+        return IExchange(exchange).withdrawBlockFee(blockIdx, feeRecipient);
+    }
+
+    /// @dev Make a withdrawal request onchain.
+    function withdraw(
+        address token,
+        uint96 amount
+        )
+        external
+        payable
+        onlyOwner
+    {
+        IExchange(exchange).withdraw(token, amount);
+    }
+
+    /// @dev Withdrawal the tokens/ether held in this contract.
+    function withdrawTo(
```

Comment:
Maybe we should allow setting a recipient address on an Account. The recipient address can only be set by the account owner. This recipient address would get all tokens from all withdrawals from that point on.  I remember Dolomite asking for this once. Could also be useful to make the contract that distributes the tokens another contract so it can be easily uprgraded/...

Reply:
will do. This PR is still very draft. Will have to think about it.

Reply:
We would need to add this to the exchange contract e.g. a `IExchange.setRecipient(address)` function that sets a `recipient` field in `Account`. If set, this recipient address would receive the ETH/tokens instead of the Account owner.